### DISCUSSION TITLE
working on why jpeg.cgi seems to clobber /configs/ for some V3s - imp_monitor.sh error log?

### DISCUSSION THREAD

**Discussion by virmaior** [gtxaspec/wz_mini_hacks#701] [Development]
I've added some debug trapping code, but I'm also wondering.

I noticed lots of this pattern of repeated errors in `daemon.log`:

```
optarg = 0:48:4:
err: param (chn:offset:size:data) err 
optarg = 0:52:4:
err: param (chn:offset:size:data) err 
optarg = 1:44:4:
err: param (chn:offset:size:data) err 
optarg = 1:48:4:
err: param (chn:offset:size:data) err 
optarg = 1:52:4:
err: param (chn:offset:size:data) err 
```

The cause being `[imp_monitor.sh](https://github.com/gtxaspec/wz_mini_hacks/blob/88dd459d5950d154f1fb9e6b20caf0bfe9a06af1/SD_ROOT/wz_mini/usr/bin/imp_monitor.sh#L36)` doesn't seem respond well to not having values set for

```

eval desired_uTargetBitRate=\$VIDEO_${i}_TARGET_BITRATE
eval desired_uMaxBitRate=\$VIDEO_${i}_MAX_BITRATE
```
on lines 18-19.

What should the default values be?

**Comment by vic4news** [comment]
I noticed this back in August. See https://github.com/gtxaspec/wz_mini_hacks/discussions/594

You can set the following in the wz_mini.config to whatever values you want to stop the parse errors. 
For example,

VIDEO_0_MAX_BITRATE="8192"
VIDEO_0_TARGET_BITRATE="4096"
VIDEO_0_ENC_PARAMETER="2"
VIDEO_0_FPS="25"

VIDEO_1_MAX_BITRATE="240"
VIDEO_1_TARGET_BITRATE="180"
VIDEO_1_ENC_PARAMETER="2"
VIDEO_1_FPS="25"

**Comment by virmaior** [comment]
can narrow that it's not `cmd jpeg` itself that clobbers using the following code:
```
px=$(cat /configs/TZ)
pa=$(date)
cmd jpeg "$GET_channel" -n
py=$(cat /configs/TZ)
pb=$(date)
echo $pa ' - ' $px " and " $pb  ' - ' $py >> /opt/wz_mini/www/cgi-bin/jpeg_log.txt
```

when the `/configs` get wiped, it's wiped ahead of cmd jpeg...

**Reply by virmaior** [reply]
but can also confirm it's happening at some point during the `jpeg.cgi` code ... adding more logging, researching.

**Comment by virmaior** [comment]
at least one of my tests indicates that it could be something in `shared.cgi` ... will look more carefully in there and add more trapping.
one thing that confuses me is the intermittency.

**Reply by gtxaspec** [reply]
@virmaior my theory is iCamera is causing the corruption.  

When we are reading from /configs, are we locking the files in anyway?  Maybe iCamera doesn't like something else reading from it... 

Perhaps we should probably not touch `/configs` at all, even reading, or, if we do need to read, maybe we should read from a copy of the config files?

**Reply by virmaior** [reply]
@gtxaspec I think that's a great idea re: if we can avoid using it. 

I'll continue working on trapping to see if there's a particular command that's doing it right now, but I do like the theory that iCamera itself is trying to access `/configs` and then it clobbers it if finds a conflict while trying to access it or other issue. It's possible that it's trying to mount/unmount when it has trouble but a file being open from a different process prevents that?

Another theory which I'll put down as my second option for now is that the flying squirrel has superpowers related to not just batteries but also to hacking and is killing the `/configs` himself. He did deactivate a friend's car battery when he didn't want to move.

**Comment by virmaior** [comment]
just put in a pull request that should make it so that the `www/cgi-bin` stuff only touches `/configs` on its very first run each boot and instead makes those environment variables for reuse at `/opt/wz_mini/tmp/www_env.sh`
hopefully this will decrease or at least help us better debug the clobbering issue.
